src/main/java/generateDatasets.java

```diff
@@ -0,0 +1,165 @@
+        dataset1.add(y1);
+        dataset.add(dataset1);
+        return dataset;
+    }
+
+    public static ArrayList<ArrayList<ArrayList<Double>>> generateClusters() {
+        ArrayList<ArrayList<ArrayList<Double>>> clusters = new ArrayList<>();
+
+        // Cluster 1 Parameters
+        double cluster1CenterX = 5.0;
+        double cluster1CenterY = 5.0;
+        double cluster1StdDeviation = 1.5;
+        int cluster1Size = 1000;
+
+        ArrayList<ArrayList<Double>> cluster1 = generateCluster(cluster1CenterX, cluster1CenterY, cluster1StdDeviation, cluster1Size);
+        clusters.add(cluster1);
+
+        // Cluster 2 Parameters
+        double cluster2CenterX = -5.0;
+        double cluster2CenterY = -5.0;
+        double cluster2StdDeviation = 1.0;
+        int cluster2Size = 1000;
+
+        ArrayList<ArrayList<Double>> cluster2 = generateCluster(cluster2CenterX, cluster2CenterY, cluster2StdDeviation, cluster2Size);
+        clusters.add(cluster2);
+
+        return clusters;
+    }
+
+    public static ArrayList<ArrayList<Double>> generateCluster(double centerX, double centerY, double stdDeviation, int size) {
+        ArrayList<ArrayList<Double>> cluster = new ArrayList<>();
+        ArrayList<Double> x = new ArrayList<>();
+        ArrayList<Double> y = new ArrayList<>();
+
+        Random random = new Random();
+
+        for (int i = 0; i < size; i++) {
+            double xValue = random.nextGaussian() * stdDeviation + centerX;
+            double yValue = random.nextGaussian() * stdDeviation + centerY;
+            x.add(xValue);
+            y.add(yValue);
+        }
+
+        cluster.add(x);
+        cluster.add(y);
+
+        return cluster;
+    }
+
+    public static ArrayList<ArrayList<ArrayList<Double>>> generateQuadrantDatasets() {
+        ArrayList<ArrayList<ArrayList<Double>>> datasets = new ArrayList<>();
+
+        // Generate the first and third quadrants dataset
+        ArrayList<ArrayList<Double>> dataset1 = generateQuadrantDataset(500, 1, 1);
+        datasets.add(dataset1);
+
+        // Generate the second and fourth quadrants dataset
+        ArrayList<ArrayList<Double>> dataset2 = generateQuadrantDataset(500, -1, -1);
+        datasets.add(dataset2);
+
+        return datasets;
+    }
+
+    public static ArrayList<ArrayList<Double>> generateQuadrantDataset(int size, int xSign, int ySign) {
+        ArrayList<ArrayList<Double>> dataset = new ArrayList<>();
+        ArrayList<Double> x = new ArrayList<>();
+        ArrayList<Double> y = new ArrayList<>();
+
+        Random random = new Random();
+
+        for (int i = 0; i < size; i++) {
+            double xValue = random.nextDouble() * xSign * 10.0;
+            double yValue = random.nextDouble() * ySign * 10.0;
+            x.add(xValue);
+            y.add(yValue);
+        }
+
+        dataset.add(x);
+        dataset.add(y);
+
+        return dataset;
+    }
+
+    public static ArrayList<ArrayList<ArrayList<Double>>> generateSpiralDatasets() {
```

Comment:
So we want our datasets looking like:

$((x,y),\hat{y})$

or maybe like (so we have no ragged tensors):

$((x,y), (\hat{y}))$

where $x$, $y$ are co-ordinates and $\hat{y}$ is our label.

How do our outputs currently look from this?

Reply:
Currently I have, [ { (x), (y) }, { (x), (y) } ].
How do you want it?

Reply:
whats the second pair of x and y for?

Reply:
Each pair of coordinates correspond to their respective weight, in this case, 0 and 1.